File: 1720-decode-xored-array/1720-decode-xored-array.js
/**
 * @param {number[]} encoded
 * @param {number} first
 * @return {number[]}
 */
var decode = function(encoded, first) {
  const result = []
  result.push(first)
  let pre = first
  for (let n of encoded) {
    pre = pre ^ n
    result.push(pre)
  }
  return result
};

File: 1720-decode-xored-array/1720-decode-xored-array.go
func decode(encoded []int, first int) []int {
  pre := first
  ans := make([]int, len(encoded) + 1)
  ans[0] = first
  for i, v := range encoded {
    pre = pre ^ v
    ans[i+1] = pre
  }  
  return ans
}